content_script.js

```diff
@@ -0,0 +1,82 @@
+var filteredTags = ["script", "noscript", "style"];//, "link"];
+
+function serializeTree(html, holes, element, depth) {
+	if (!element.tagName && element.nodeType != 3) {
+		// ignore elements that don't have tags and are not text
+	}
+	else if (element.tagName && filteredTags.indexOf(element.tagName.toLowerCase()) != -1) {
+		// filter out elements that are in filteredTags
+	}
+	else if (element.nodeType == 3) {
+		html.push(element.textContent);
+	}
+	else {
+		html.push(new Array(depth+1).join("  "));
+		html.push("<" + element.tagName + " ");
```

Comment:
Consider whether an ES6 template literal would be easier to read:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals